dev-lang/vala/vala-0.40.3.recipe

```diff
@@ -0,0 +1,106 @@
+SUMMARY="An object-oriented programming language that uses the GObject system"
+DESCRIPTION="Vala is a new programming language that aims to bring modern \
+programming language features to GNOME developers without imposing any \
+additional runtime requirements and without using a different ABI \
+compared to applications and libraries written in C."
+HOMEPAGE="https://wiki.gnome.org/Projects/Vala"
+COPYRIGHT="2006-2018 Jürg Billeter
+	2006-2018 Raffaele Sandrini"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://download.gnome.org/sources/vala/${portVersion%.*}/vala-$portVersion.tar.xz"
+CHECKSUM_SHA256="6d4f1f0b2edaa0d4aa96f72735a9845e6b1acf80a0a2ae494b5d43e07833119e"
+SOURCE_DIR="vala-$portVersion"
+PATCHES="vala-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+libVersion="0.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+portVers="${portVersion%.*}"
+
+PROVIDES="
+	vala$secondaryArchSuffix = $portVersion
+	cmd:vala$secondaryArchSuffix
+	cmd:vala_$portVers$secondaryArchSuffix
+	cmd:vala_gen_introspect$secondaryArchSuffix
+	cmd:vala_gen_introspect_$portVers$secondaryArchSuffix
+	cmd:valac$secondaryArchSuffix
+	cmd:valac_$portVers$secondaryArchSuffix
+	cmd:valadoc$secondaryArchSuffix
+	cmd:valadoc_$portVers$secondaryArchSuffix
+	cmd:vapigen$secondaryArchSuffix
+	cmd:vapigen_$portVers$secondaryArchSuffix
+	lib:libvala_$portVers$secondaryArchSuffix = $libVersionCompat
+	lib:libvaladoc_$portVers$secondaryArchSuffix = $libVersionCompat
+	devel:libvala_$portVers$secondaryArchSuffix = $libVersionCompat
+	devel:libvaladoc_$portVers$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcdt$secondaryArchSuffix
+	lib:libcgraph$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgvc$secondaryArchSuffix
```

Comment:
```diff
 	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgmodule_2.0$secondaryArchSuffix
+	lib:libgobject_2.0$secondaryArchSuffix
 	lib:libgvc$secondaryArchSuffix
```
This is not strictly required because `libgobject-2.0` and `libgmodule-2.0` are being pulled by `libglib-2.0`.
But it is better to mention these libs because they are needed by `lib/valadoc/doclets/*/*.so`.